tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1616,6 +1587,245 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
+									return false, nil
+								}
+								pvcUIDs[pvc.Name] = string(pvc.UID)
+							}
+						}
+						return true, nil
+					})
+					framework.ExpectNoError(err, "expect PVCs of scaled in Pods to have annotation tidb.pingcap.com/pvc-defer-deleting")
+
+					ginkgo.By(fmt.Sprintf("Scale out %s", comp))
+					err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+						if comp == "PD" {
+							tc.Spec.PD.Replicas = 5
+						} else if comp == "TiKV" {
+							tc.Spec.TiKV.Replicas = 4
+						} else if comp == "TiDB" {
+							tc.Spec.TiDB.Replicas = 3
+						}
+						return nil
+					})
+					framework.ExpectNoError(err, "failed to scale out %s for TidbCluster %s/%s", comp, ns, tc.Name)
+					ginkgo.By(fmt.Sprintf("Wait for %s to be in ScalePhase", comp))
+					if comp == "PD" {
+						utiltc.MustWaitForPDPhase(cli, tc, v1alpha1.ScalePhase, 3*time.Minute, 10*time.Second)
+					} else if comp == "TiKV" {
+						utiltc.MustWaitForTiKVPhase(cli, tc, v1alpha1.ScalePhase, 3*time.Minute, 10*time.Second)
+					} else if comp == "TiDB" {
+						utiltc.MustWaitForTiDBPhase(cli, tc, v1alpha1.ScalePhase, 3*time.Minute, 10*time.Second)
+					}
+					log.Logf(fmt.Sprintf("%s is in ScalePhase", comp))
+					ginkgo.By("Wait for tc ready")
+					err = oa.WaitForTidbClusterReady(tc, 3*time.Minute, 10*time.Second)
+					framework.ExpectNoError(err, "failed to wait for TidbCluster %s/%s ready after scale out %s", ns, tc.Name, comp)
+
+					ginkgo.By(fmt.Sprintf("Check PVCs are recreated for newly scaled out %s", comp))
+					for ordinal := 3; ordinal < 5; ordinal++ {
+						var pvcSelector labels.Selector
+						if comp == "PD" {
+							pvcSelector, err = member.GetPVCSelectorForPod(tc, v1alpha1.PDMemberType, int32(ordinal))
+						} else if comp == "TiKV" {
+							pvcSelector, err = member.GetPVCSelectorForPod(tc, v1alpha1.TiKVMemberType, int32(ordinal))
+						} else if comp == "TiDB" {
+							pvcSelector, err = member.GetPVCSelectorForPod(tc, v1alpha1.TiDBMemberType, int32(ordinal))
+						}
+						framework.ExpectNoError(err, "failed to get PVC selector for tc %s/%s", tc.GetNamespace(), tc.GetName())
+						pvcs, err := c.CoreV1().PersistentVolumeClaims(ns).List(metav1.ListOptions{LabelSelector: pvcSelector.String()})
+						framework.ExpectNoError(err, "failed to list PVCs with selector: %v", pvcSelector)
+						for _, pvc := range pvcs.Items {
+							annotations := pvc.GetObjectMeta().GetAnnotations()
+							log.Logf("pvc annotations: %+v", annotations)
+							_, ok := annotations["tidb.pingcap.com/pvc-defer-deleting"]
+							framework.ExpectEqual(ok, false, "expect PVC %s/%s not to have annotation tidb.pingcap.com/pvc-defer-deleting", pvc.GetNamespace(), pvc.GetName())
+							pvcUIDString := pvcUIDs[pvc.Name]
+							framework.ExpectNotEqual(string(pvc.UID), pvcUIDString)
+						}
+					}
+				})
+			}
+		})
+
+		ginkgo.Context("while concurrently upgrade should work for", func() {
```

Comment:
Please also help cover the TiDB and TiKV.

Reply:
covered